### DISCUSSION TITLE
Does `pure` allow returning string (or arrays of strings)?

### DISCUSSION THREAD

**Discussion by keithlayne** [statelyai/xstate#1339] [General] [answered]
In TS this seems to be a type error.  My assumption was that a named action would be a valid return from `pure` - and when I throw a type assertion on the return value, it seems to work as I expect.  Is this just an oversight in the typings?

**Comment by davidkpiano** [comment, marked as answer]
It expects: `SingleOrArray<ActionObject<TContext, TEvent>> | undefined`

The workaround here is to wrap it in an object: `{ type: 'yourAction' }`, which is more explicit anyway.

**Reply by keithlayne** [reply]
Yeah that actually didn't work, just getting around to trying it, but I don't think your suggestion is the problem :(

**Reply by davidkpiano** [reply]
Feel free to share a CodeSandbox if you'd like more help.
